### DISCUSSION TITLE
How to set DDC max to 255 to solve Apple Thunderbolt display 30% brightness issue?

### DISCUSSION THREAD

**Discussion by Mtkhan13** [MonitorControl/MonitorControl#1637] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

I have used this latest app on my mackintosh with Apple Thunderbolt display. But now the brightness is stuck at 30% it doesn't go up-to 100% even the app controls are at 100%. Please let me know what is the problem and how can I fix it. 
Thanks

### Expected behavior

At 100% range the brightness should be 100% but not 30%. 

### Anything else?

https://github.com/MonitorControl/MonitorControl/discussions/739 I have read this comment as well its says, it can be change in the code to 250 and that works fine. But I tried downloading few builds but I'm getting so many error while running the code. 
![image](https://github.com/user-attachments/assets/b141587e-8a26-407d-80cc-575a6eb1c844)
![image](https://github.com/user-attachments/assets/13681985-05c4-40e8-b4ed-2d811363d80e)

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.7.10
- Mac model: Hackintosh iMac Mid 2015
- MonitorControl version: 4.3.2
- Monitor(s): Apple Thunderbolt Display 27"
- Monitor Cable(s)/Connection(s): iMac (HP Thunderbolt 2 Pcie) -> Thunderbolt 2 -> Apple thunderbolt Display 27"
- Apple Silicon/M1 (yes or no): No
```

**Comment by waydabber** [comment, marked as answer]
Hmm. Not sure why are you trying to recompile the app. Simply go to Settings/Displays/Show advanced (bottom) and set DDC max to 255.

**Comment by Mtkhan13** [comment]
Hi, thanks for reply. Yes I have already tried that as well but it didn't work. I set it 250 and slide the brightness but no luck. --Sent with mail.com Mail appOn 10/3/24, 10:14 AM waydabber ***@***.***> wrote:

  
  Hmm. Not sure why are you trying to recompile the app. Simply go to Settings/Displays/Show advanced (bottom) and set DDC max to 255.

**Reply by waydabber** [reply]
In that case I am not sure what the problem is sadly. You can try installing BetterDisplay and use DDC Factory Reset (first you need to force-set the display to third party display as BetterDisplay by default does not allow DDC control for older Apple displays precisely to prevent the issue you encountered).

**Reply by Mtkhan13** [reply]
I just temporarily need to use this app, because in my Hackintosh I've not yet figured out the native controls to work. So at 100% brightness the display gets too hot and I'm sure If I will continue using it that way, its definitely gonna burn at some point so I'm figuring out the solution. Meanwhile I saw your app and thought to give it a try. Although it reduced too much of the brightness but seems the temperature is quite normal now. Your are app is working fine on my 2nd HP display attached to same Hackintosh but its not working on the thunderbolt apple display. 

You are suggesting to use the BetterDisplay but please guide me how to reset the DDC?

thanks a lot for the guidance and support!

**Reply by waydabber** [reply]
Normally Apple Thunderbolt display is not designed to work with DDC. In BetterDisplay you should go to the DDC Control menu and select Factory Reset. If the DDC menus are not available, go to Settings/Displays and under one of the advanced settings unfoldable block there is a "Treat as third party display" or something similar, you should enable that so DDC is available (it is not enabled for Apple made displays by default).

**Reply by Mtkhan13** [reply]
It seems like your app has made changes permanently on my thunderbolt display. I tried BetterDisplay but still no luck brightness is max at 30% I can't get it back to 100% thats really frustrating.  Please guide me how to rest my display. 
Appreciate it, Thanks

**Comment by Mtkhan13** [comment]
That's didn't work at all. I tried the 3rd party option and DDC rest but no luck. --Sent with mail.com Mail appOn 10/3/24, 6:40 PM waydabber ***@***.***> wrote:

  
  Normally Apple Thunderbolt display is not designed to work with DDC. In BetterDisplay you should go to the DDC Control menu and select Factory Reset. If the DDC menus are not available, go to Settings/Displays and under one of the advanced settings unfoldable block there is a "Treat as third party display" or something similar, you should enable that so DDC is available (it is not enabled for Apple made displays by default).

**Reply by waydabber** [reply]
Well, then your only option seems to be to connect it to a real Mac that supports hardware control via Thunderbolt I am afraid. Normally most other users seem to be able to make this work by properly changing the DDC max, not sure why yours does not work, maybe it has a different firmware. I recommend looking through the many discussions about this as many had the same issue and could fix it, but I did not read through all discussions. I actually don't have first hand experience with this display.
